log/stat/stat.go

```diff
@@ -0,0 +1,117 @@
+package stat
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+	"text/tabwriter"
+
+	"github.com/peak/s5cmd/strutil"
+)
+
+const (
+	totalCount = iota
+	succCount
+)
+
+var (
+	enabled bool
+	stats   statistics
+)
+
+type statistics [2]syncMap
+
+// InitStat initializes collecting program statistics.
+func InitStat() {
+	enabled = true
+	for i := range stats {
+		stats[i] = syncMap{
+			Mutex:       sync.Mutex{},
+			mapStrInt64: map[string]int64{},
+		}
+	}
+}
+
+// syncMap is a statically typed and synchronized map.
+type syncMap struct {
+	sync.Mutex
+	mapStrInt64 map[string]int64
+}
+
+func (s *syncMap) add(key string, val int64) {
+	s.Lock()
+	defer s.Unlock()
+
+	s.mapStrInt64[key] += val
+}
+
+// Stat is for storing a particular statistics.
+type Stat struct {
+	Visited    string `json:"operation"`
+	SuccVisits int64  `json:"success"`
+	ErrVisits  int64  `json:"error"`
+}
+
+// Collect collects function execution data.
+func Collect(path string, err *error) func() {
+	return func() {
+		if !enabled {
+			return
+		}
+		if err == nil || *err == nil {
+			stats[succCount].add(path, 1)
+		}
+		stats[totalCount].add(path, 1)
+	}
+}
+
+// Stats implements log.Message interface.
+type Stats []Stat
+
+func (s Stats) String() string {
+	if len(s) == 0 {
+		return ""
```

Comment:
Same comments as below, if it just prints headers without data, it can be treated as TSV.